### PR TITLE
[FLINK-17937][hive] Change some hive connector tests to IT cases

### PR DESCRIPTION
**Pull request by lirui-apache**
## What is the purpose of the change

Hive connector tests that use FlinkStandaloneHiveRunner should be IT cases. Besides, changing them to IT cases can avoid lib conflicts because we'll be using the shaded jars, e.g. guava.

## Brief change log

  - Change some tests to IT cases
  - Fix `HiveTableSourceITCase` so that it runs with hive 1.x
  - Update `HiveLookupJoinTest` so that it doesn't require the hive runner

## Verifying this change

Existing tests

## Does this pull request potentially affect one of the following parts:

N/A

## Documentation

N/A

### CODE REVIEW

**Comment by lirui-apache** [comment]
cc @JingsongLi

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveLookupJoinTest.java [hunk 1]
```diff
@@ -50,7 +48,6 @@
 /**
  * Test lookup join of hive tables.
  */
-@RunWith(FlinkStandaloneHiveRunner.class)
 public class HiveLookupJoinTest {
```
Comment (JingsongLi) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveLookupJoinTest.java:51]: ITCase

**Comment by lirui-apache** [comment]
@JingsongLi Thanks for the review. Updated accordingly.

**Comment by lirui-apache** [comment]
The failure shouldn't be related. My personal pipeline has passed: https://dev.azure.com/lirui-apache/flink/_build/results?buildId=96&view=results

**Pull request merged**